deployment/terraform/create.go

```diff
@@ -392,6 +434,166 @@ func (t *Terraform) setupLoadtestAgents(extAgent *ssh.ExtAgent, initData bool) e
+	if err != nil {
+		return fmt.Errorf("unable to list indices: %w", err)
+	}
+	mlog.Debug("Indices in ElasticSearch domain", mlog.Array("indices", indices))
+
+	repositories, err := es.ListRepositories()
+	if err != nil {
+		return fmt.Errorf("unable to list repositories: %w", err)
+	}
+	mlog.Debug("Repositories registered", mlog.Array("repositories", repositories))
+
+	repo := t.config.ElasticSearchSettings.SnapshotRepository
+
+	// Check if the registered repositories already include the one configured
+	repoFound := false
+	for _, r := range repositories {
+		if r.Name == repo {
+			repoFound = true
+			break
+		}
+	}
+
+	// Register the repository configured if not found
+	if !repoFound {
+		arn := output.ElasticSearchRoleARN
+		if err := es.RegisterS3Repository(repo, arn); err != nil {
+			return fmt.Errorf("unable to register repository: %w", err)
+		}
+		mlog.Info("Repository registered", mlog.String("repository", repo))
+	}
+
+	// List all snapshots in the configured repository
+	snapshots, err := es.ListSnapshots(repo)
+	if err != nil {
+		return fmt.Errorf("unable to list snapshots: %w", err)
+	}
+	mlog.Debug("Snapshots in registered repository", mlog.Array("snapshots", snapshots))
+
+	// Look for the configured snapshot
+	var snapshot elasticsearch.Snapshot
+	snapshotName := t.config.ElasticSearchSettings.SnapshotName
+	for _, s := range snapshots {
+		if s.Name == snapshotName {
+			snapshot = s
+			break
+		}
+	}
+	if snapshot.Name == "" {
+		return fmt.Errorf("snapshot %q not found in repository %q", snapshotName, repo)
+	}
+
+	// Filter out indices starting with '.', like .kibana_1
+	snapshotIndices := []string{}
+	for _, i := range snapshot.Indices {
+		if !strings.HasPrefix(i, ".") {
+			snapshotIndices = append(snapshotIndices, i)
+		}
+
+	}
+	mlog.Debug("Indices in the snapshot to be restored", mlog.Array("snapshot indices", snapshotIndices))
+
+	// We need to close all indices already present in the ElasticSearch
+	// server that we want to restore from the snapshot. This is needed for
+	// fresh servers as well, since at least the users index is sometimes
+	// present on creation.
+	indicesToClose := []string{}
+	for _, index := range indices {
+		for _, snapshotIndex := range snapshotIndices {
+			if index == snapshotIndex {
+				indicesToClose = append(indicesToClose, index)
+			}
+		}
+	}
+
+	if len(indicesToClose) > 0 {
+		mlog.Info("Closing indices in ElasticSearch server...", mlog.Array("indices", indicesToClose))
+		if err := es.CloseIndices(indicesToClose); err != nil {
+			return fmt.Errorf("unable to close indices: %w", err)
+		}
+	}
+
+	mlog.Info("Restoring snapshot...",
+		mlog.String("repository", repo),
+		mlog.String("snapshot", snapshotName),
+		mlog.Array("indices", snapshotIndices))
+	opts := elasticsearch.RestoreSnapshotOpts{
+		WithIndices: snapshotIndices,
+	}
+	if err := es.RestoreSnapshot(repo, snapshotName, opts); err != nil {
+		return fmt.Errorf("unable to restore snapshot: %w", err)
+	}
+
```

Comment:
It will re-restore the snapshot, closing all indices in the Elasticsearch server and then restoring them from the snapshot. This is what I was testing in my comments above ([this one](https://github.com/mattermost/mattermost-load-test-ng/pull/725#issuecomment-2050348060) and the ones before): the process works, but I noticed that the number of documents slightly decreased in the second run of `deployment create`.